### PR TITLE
-Encoding add ANSI option

### LINKED ISSUE #6562: Need "ANSI" encoding enumeration value to support "ANSI"-code-page-encoded files (e.g., Windows 1252)
As discussed in #6550:

While you can pass `OEM` to filesystem cmdlets to support the legacy system locale's OEM code page on Windows, its "ANSI" counterpart (such as Windows 1252 on US-English systems) is currently missing.
(In _Windows PowerShell_, the `Default` value fulfills that role, but in PowerShell Core `Default` now refers to the new default, (BOM-less) UTF-8.)

Therefore, an `ANSI` encoding value should be introduced to complement the `OEM` value.

With `ANSI` available, the current workaround:

```powershell
Get-Content -Encoding ([cultureinfo]::CurrentCulture.TextInfo.ANSICodePage) file.txt
```

would simply become:

```powershell
Get-Content -Encoding Ansi file.txt   # Wishful thinking.
```

Note: Given that `OEM` already is available even when running on Unix-like platforms, it sounds like we shouldn't restrict `ANSI`'s availability to Windows. (`[System.Text.Encoding]::GetEncoding([cultureinfo]::CurrentCulture.TextInfo.ANSICodePage)`seemingly does return a locale-appropriate value on Unix-like platforms as well.)

Environment data
----------------

Written as of:

```powershell
PowerShell Core v6.0.2
```

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary

- Add ANSI option to -Encoding
- Small cleanup
- Reorder stuff

-Encoding ANSI equals to `[System.Text.Encoding]::GetEncoding([CultureInfo]::CurrentCulture.TextInfo.ANSICodePage)`
or `-Encoding [CultureInfo]::CurrentCulture.TextInfo.ANSICodePage`

## PR Context
Fixes https://github.com/PowerShell/PowerShell/issues/6562 @mklement0 

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/9887
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by mklement0** [comment]
Thanks for tackling this, @CarloToso.

**Review by iSazonov — COMMENTED** [review]
Please add tests

File: src/System.Management.Automation/utils/EncodingUtils.cs [hunk 1]
```diff
@@ -10,25 +11,26 @@ namespace System.Management.Automation
...
-        internal const string Utf8NoBom = "utf8NoBOM";
         internal const string Utf8Bom = "utf8BOM";
-        internal const string Utf7 = "utf7";
+        internal const string Utf8NoBom = "utf8NoBOM";
         internal const string Utf32 = "utf32";
-        internal const string Default = "default";
-        internal const string OEM = "oem";
+        internal const string ANSI = "ansi";
+        internal const string String = "string";
+        internal const string Unknown = "unknown";
```
[the remark this answers is not in the document]
Comment (CarloToso) [reply src/System.Management.Automation/utils/EncodingUtils.cs:27]: I ordered them ABC (TabCompletionResults) + ABC (Others), is it ok?

[the remark this answers is not in the document]
Comment (iSazonov) [reply src/System.Management.Automation/utils/EncodingUtils.cs:27]: It is one list of constants so I'd sort it as whole.

[the remark this answers is not in the document]
Comment (CarloToso) [reply src/System.Management.Automation/utils/EncodingUtils.cs]: done

**Comment by CarloToso** [comment]
Unrelated error

**Review by iSazonov — APPROVED** [review]

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
This change is adding to an already custom list, so I don't believe there's a concern supporting ANSI encoding

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/utils/EncodingUtils.cs [hunk 2]
```diff
@@ -10,41 +11,43 @@ namespace System.Management.Automation
...
-        internal const string Utf8NoBom = "utf8NoBOM";
         internal const string Utf8Bom = "utf8BOM";
-        internal const string Utf7 = "utf7";
+        internal const string Utf8NoBom = "utf8NoBOM";
         internal const string Utf32 = "utf32";
-        internal const string Default = "default";
-        internal const string OEM = "oem";
 
         internal static readonly string[] TabCompletionResults = {
                 Ascii, BigEndianUnicode, BigEndianUtf32, OEM, Unicode, Utf7, Utf8, Utf8Bom, Utf8NoBom, Utf32
```
[the remark this answers is not in the document]
Comment (CarloToso) [reply src/System.Management.Automation/utils/EncodingUtils.cs]: done

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**